### PR TITLE
fix(vcenter_parser): Add route for vCenter json parsing

### PR DESCRIPTION
**Pull request by kuiperda**
## Description of Changes
Adds a route to parse json coming from vCenter, based off [#243](https://github.com/observIQ/stanza-plugins/pull/243)

## **Please check that the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] CI passes

### CODE REVIEW

File: plugins/vmware_vcenter.yaml [hunk 1]
```diff
@@ -106,3 +108,46 @@ pipeline:
...
+    - move:
+        from: '$record["@fields.alias"]'
+        to: '$record.alias'
+    - move:
+        from: '$record["@fields.company_name"]'
+        to: '$record.company_name'
+    - move:
+        from: '$record["@fields.region_id"]'
+        to: '$record.region_id'
+    output: {{.output}}
```
Comment (jsirianni) [inline plugins/vmware_vcenter.yaml:153]: Nit: missing trailing newline char.

**Review by jsirianni — COMMENTED** [review]
Looks good. Unless you have already, we should test the following cases:
1. Our vcenter server still works with RFC5424
2. Example GCVE json logs work without error.

Generally I will do something like this to test TCP based logs (assuming port 5142):
```
echo '{"timestamp": "2023-07-14T11:26:45.477", "severity": "info", "message' | nc localhost 5142
```
You can replace the json log I am using with one of the example GCVE logs that we have.

File: plugins/vmware_vcenter.yaml [hunk 2]
```diff
@@ -106,3 +108,47 @@ pipeline:
...
+
+  - id: gcve_json_parser
+    type: json_parser
+    parse_from: $record
+    severity:
+      parse_from: '$record["@fields.severity"]'
+    timestamp:
+      layout: "2006-01-02T15:04:05.000000-07:00"
+      layout_type: gotime
+      parse_from: '@timestamp'
```
Comment (jsirianni) [inline plugins/vmware_vcenter.yaml:120]:
```suggestion
      parse_from: '$record["@timestamp"]'
```

File: plugins/vmware_vcenter.yaml [hunk 3]
```diff
@@ -106,3 +108,47 @@ pipeline:
...
+        from: '$record["@fields.alias"]'
+        to: '$record.alias'
+    - move:
+        from: '$record["@fields.company_name"]'
+        to: '$record.company_name'
+    - move:
+        from: '$record["@fields.region_id"]'
+        to: '$record.region_id'
+    output: {{.output}}
+
```
Comment (jsirianni) [inline plugins/vmware_vcenter.yaml:154]: This is usually resolvable with a setting in your editor. I know vscode has an option to include new line characters on save.

**Review by jsirianni — APPROVED** [review]

File: plugins/vmware_vcenter.yaml [hunk 4]
```diff
@@ -106,3 +108,47 @@ pipeline:
...
```
Comment (schmikei) [inline plugins/vmware_vcenter.yaml:154]: Is this newline thing still an issue? Or just a github artifact?

Comment (kuiperda) [reply plugins/vmware_vcenter.yaml:154]: I think it's fine

**Review by schmikei — APPROVED** [review]

**Pull request merged**